src/main/java/ch/njol/skript/expressions/ExprPassengers.java

```diff
@@ -0,0 +1,181 @@
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.bukkit.event.vehicle.VehicleEnterEvent;
+import org.bukkit.event.vehicle.VehicleExitEvent;
+import org.eclipse.jdt.annotation.Nullable;
+import org.skriptlang.skript.lang.converter.Converter;
+import org.spigotmc.event.entity.EntityDismountEvent;
+import org.spigotmc.event.entity.EntityMountEvent;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.EventValues;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Passengers")
+@Description({
+	"The passenger of a vehicle, or the rider of a mob.",
+	"For 1.11.2 and above, it returns a list of passengers and you can use all changers in it.",
+	"See also: <a href='#ExprVehicle'>vehicle</a>"
+})
+@Examples({
+	"#for 1.11 and lower",
+	"passenger of the minecart is a creeper or a cow",
+	"the saddled pig's passenger is a player",
+	"#for 1.11.2+",
+	"passengers of the minecart contains a creeper or a cow",
+	"the boat's passenger contains a pig",
+	"add a cow and a zombie to passengers of last spawned boat",
+	"set passengers of player's vehicle to a pig and a horse",
+	"remove all pigs from player's vehicle",
+	"clear passengers of boat"
+})
+@Since("2.0, 2.2-dev26 (Multiple passengers for 1.11.2+)")
+public class ExprPassengers extends SimpleExpression<Entity> { // SimpleExpression due to isSingle
+
+	static {
+		PropertyExpression.registerDefault(ExprPassengers.class, Entity.class, "passenger[:s]", "entities");
+	}
+
+	@Nullable
+	private Expression<Entity> vehicles;
+	private boolean plural;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		vehicles = (Expression<Entity>) exprs[0];
+		plural = parseResult.hasTag("s");
+		return true;
+	}
+
+	@Override
+	@Nullable
+	protected Entity[] get(Event event) {
+		Entity[] source = vehicles.getArray(event);
+		Converter<Entity, Entity[]> converter = new Converter<Entity, Entity[]>() {
+			@Override
+			@Nullable
+			public Entity[] convert(Entity entity) {
+				if (getTime() != EventValues.TIME_PAST && event instanceof VehicleEnterEvent && entity.equals(((VehicleEnterEvent) event).getVehicle()))
```

Comment:
Can we initialize a variable with `getTime() != EventValues.TIME_PAST` instead of writing that statement multiple times?